geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala

```diff
@@ -0,0 +1,174 @@
+      getChunkBounds(geoHash, bbox, gridStepX, gridStepY).foreach { case (minX, maxX, minY, maxY) =>
+        if (maxX > minX && maxY > minY) {
+          val cmd = Seq("gdal_translate", "-q", "-of", "GTIFF", "-projwin", s"${minX}", s"${maxY}",
+            s"${maxX}", s"${minY}", file, s"${outDir}/gh_${geoHash.hash}.tif")
+          cmd.!!
+        }
+      }
+    }
+  }
+
+  //Get bounds from given GeoHash that are used to crop raster
+  def getChunkBounds(gh: GeoHash,
+                     bbox: BoundingBox,
+                     stepX: Double,
+                     stepY: Double): Option[(Double, Double, Double, Double)] = {
+    BoundingBox.intersects(bbox, gh.bbox) match {
+      case false => None
+      case _ =>
+        val (ghMinX, ghMinY, ghMaxX, ghMaxY) = (gh.bbox.ll.getX, gh.bbox.ll.getY, gh.bbox.ur.getX, gh.bbox.ur.getY)
+        val (rMinX, rMinY, rMaxX, rMaxY) = (bbox.ll.getX, bbox.ll.getY, bbox.ur.getX, bbox.ur.getY)
+        Some(getCropLocation(ghMinX, rMinX, rMaxX, stepX, false),
+             getCropLocation(ghMaxX, rMinX, rMaxX, stepX, true),
+             getCropLocation(ghMinY, rMinY, rMaxY, stepY, false),
+             getCropLocation(ghMaxY, rMinY, rMaxY, stepY, true))
+    }
+  }
+
+  //Get location on raster grid for cropping operation
+  def getCropLocation(coor: Double,
+                      minCoor: Double,
+                      maxCoor: Double,
+                      step: Double,
+                      isMax: Boolean): Double = {
+    val steps = ((coor - minCoor) / step).toInt
+    val calCoor = {
+      val tmpCoor = minCoor + step * steps.toDouble
+      //NO DUPLICATION: With the following line, adjacent chunks have no overlapped edge.
+      if (coor - tmpCoor > 0.5D * step) tmpCoor + step
+      //DUPLICATION POSSIBLE: With the following line, adjacent chunks may have one edge
+      //(width = 1) overlapped.
+      //if (isMax && coor - tmpCoor > 0D) tmpCoor + step
+      else tmpCoor
+    }
+    Math.min(Math.max(calCoor, minCoor), maxCoor)
+  }
+}
+
+object RasterChunking extends Logging {
+  val BOUND_MIN_X = -180.0
+  val BOUND_MIN_Y = -90.0
+  val BOUND_MAX_X = 180.0
+  val BOUND_MAX_Y = 90.0
+
+  //Get precision of geohash with size no larger than given bound
+  def GetGeoHashPrecisionBySize(rasterGrid: GridCoverage2D, size: Double): Int = {
+    val envelope = rasterGrid.getEnvelope2D
+    val (stepX, stepY) = getSteps(rasterGrid)
+    val pixelSize = getPixelSize(rasterGrid)
+    val bbox = BoundingBox(envelope.getMinX, envelope.getMaxX, envelope.getMinY, envelope.getMaxY)
+    var gh = GeohashUtils.getClosestAcceptableGeoHash(bbox.geom.getEnvelopeInternal).getOrElse(
+      throw new Exception("No closest acceptable Geohash found. Cannot chunk raster."))
+
+    while (getGridSize(gh, stepX, stepY) * pixelSize / 1000.0 > size) {
+      gh = GeoHash(s"${gh.hash}") //There was a "0" in the string after the hash, was this a typo?
```

Comment:
This is a odd comment---if there is a problem make a ticket otherwise remove it.